zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -492,8 +570,9 @@ class NovaComputeActionTest(test_utils.OpenStackBaseTest):
 
     def test_virsh_audit_action(self):
         """Test virsh-audit action."""
-        for unit in zaza.model.get_units('nova-compute',
-                                         model_name=self.model_name):
+        for unit in zaza.model.get_units(
+                self.application_name,
```

Comment:
are you sure that self.application_name is being set to the name of the application in the same way as get_app_names_for_charm()?

Reply:
it is somewhat similar. The base class code that assigns self.application_name checks if each app starts with the charm_name declared in tests.yaml, and it breaks when it finds the first one. We could remove the break and go on creating a "self.application_nameS" variable and we ultimately have the same as get_app_names_for_charm. Ultimately I'd prefer if it uses "in (contains)" instead of startswith though.